src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -492,10 +499,8 @@ def _bracket_sensitive_look_ahead_match(
         """
         # Type munging
         matchers = list(matchers)
-        if isinstance(segments, BaseSegment):  # pragma: no cover TODO?
-            segments = [segments]
 
-        # Have we been passed an empty list?
+        # Have we been passed an empty tuple?
         if len(segments) == 0:
```

Comment:
Performance nitpick -- this will likely be faster (also PEP-8 compliant, not that I am a stickler for that):
```suggestion
        if not segments:
```

Reply:
👍